src/TerraformValidateStage.groovy

```diff
@@ -13,6 +14,11 @@ class TerraformValidateStage implements Stage {
         this.decorations = new HashMap<String,Closure>()
     }
 
+    public TerraformValidateStage withVersion12Support() {
```

Comment:
2) Can we be more clever with how we handle versions?  `withVersion12Support` sets an unpleasant precedence - what  happens if version 0.13 creates breaking changes?  Do we add a  `withVersion13Support`?  Ack.  What this issue reveals is a poor choice on my part - https://github.com/manheim/terraform-pipeline/blob/0233f1e41ab83d442e7f8736b41621b19ca81750/src/TerraformValidateCommand.groovy#L62.  I should never have REQUIRED that `-check-variables=false` be part of the TerraformValidateCommand. <-- During  the initial development of this library, I made several of these poor choices.  What I did later, was to break this behavior out into "default plugins", and then later expand on the plugin's behavior. 

If I were to try to take that approach for this issue, I think maybe we need a `TerraformPlugin`.  Maybe if such a Plugin existed, it could have  a `withVersion()` method, we could then isolate all the version-specific behavior in one place.

* `TerraformPlugin.init()` - would implement the existing behavior across the appropriate plugins (using pre-0.12 flags like -check-variables)
* `TerraformPlugin.withVersion('0.12').init()` - would *NOT*  add `check-variables`
* `TerraformPlugin.withVersion('FutureBreakingVersion').init()` - would do other things that specific to that future breaking version.

^-- We could even be clever, and take the solution above, and skip and explicit call  to `withVersion`.  What if we checked for the presence of a `.terraform-version` file?  If such a file existed, we could automatically call `withVersion` with the appropriate value.  Then terraform-pipeline would auto-configure for the appropriate version without the user needing to do anything at all.

Reply:
I have no problem looking for a `.terraform-version` file, I use tfenv myself.  I just didn't want to inflict the pattern on anyone else.

I believe a default plugin might work and could handle both `withVersion(String)` and detecting from the `.terraform-version`.